dashboard/app/models/programming_environment.rb

```diff
@@ -132,10 +132,24 @@ def summarize_for_index
   end
 
   def categories_for_navigation
-    categories.select(&:should_be_in_navigation?).map(&:summarize_for_navigation)
+    Rails.cache.fetch("#{cache_key_with_version}/categories_for_navigation", force: !Script.should_cache?) do
```

Comment:
whoa, TIL: https://api.rubyonrails.org/classes/ActiveRecord/Integration.html#method-i-cache_key_with_version 

was there a particular reason you used this here rather than just the `name` like you did below in `get_from_cache`? If so, I think that could be helpful to note in the comments.

Reply:
Oh not really -- it actually doesn't help much here so I might just go back to name. I was just trying it out while reading the docs